inf-clojure.el

```diff
@@ -843,6 +848,139 @@ HOST is the host the process is running on, PORT is where it's listening."
   (interactive "shost: \nnport: ")
   (inf-clojure (cons host port)))
 
+(defvar-local inf-clojure-socket-callback nil
+  "Used to transfer state between the socket process and the
+inf-clojure-connect process/buffer.")
+
+(defvar-local inf-clojure-socket-buffer nil
+  "Used to kill the socket buffer for a Clojure REPL with the REPL buffer is shutdown.")
+
+(defun inf-clojure-socket-filter (process output)
+  "A filter that gets triggered each time the socket receives new OUTPUT.
+This function prints out the output received but also
+watches for a prompt using the `inf-clojure-prompt' regexp, once
+this happens a callback is triggered if available.  The callback
+is intended to be used to trigger a `inf-clojure-connect' once we
+can determine that a socket REPL is ready to receive a
+connection.
+
+PROCESS is the process object that is being filtered.
+
+OUTPUT is the latest data received from the process"
+  (let ((server-buffer (process-buffer process)))
+    (when (buffer-live-p server-buffer)
+      (with-current-buffer server-buffer
+        (insert output)))
+    (let ((prompt-displayed (string-match inf-clojure-prompt output)))
+      (when prompt-displayed
+	(message (format "REPL prompt found for %s" (process-name process)))
+	(with-current-buffer server-buffer
+	  (when inf-clojure-socket-callback
+	    (funcall inf-clojure-socket-callback)))))))
+
+(defun inf-clojure-repl-sentinel (process event)
+  "This function is called on update change, it is mainly used to ensure the socket a REPL was connected to is cleaned up when the REPL buffer is closed."
+  (when (not (process-live-p process))
+    (let ((repl-buffer (process-buffer process)))
+      (with-current-buffer repl-buffer
+        (when inf-clojure-socket-buffer
+          (kill-buffer inf-clojure-socket-buffer))))))
+
+(defvar inf-clojure-socket-forms
+  `((lein . (concat "JVM_OPTS='-Dclojure.server.repl="
+                    "{:port %s :accept clojure.core.server/repl}' lein repl"))
+    (boot . (concat "export BOOT_JVM_OPTIONS='-Dclojure.server.repl="
+                    "\"{:port %s :accept clojure.core.server/repl}\"' boot repl"))
+    (clojure . ,(concat
+                 "clojure"
+                 " -J-Dclojure.server.repl="
+                 "\"{:port %s :accept clojure.core.server/repl}\""))
+    (cljs . ,(concat
+              "clojure"
+              " -J-Dclojure.server.repl="
+              "\"{:port %s :accept clojure.core.server/repl}\""))
+    (lein-clr . ,(concat "JVM_OPTS='-Dclojure.server.repl="
+                         "{:port %s :accept clojure.core.server/repl}' lein clr repl"))
+    (planck . "planck -n %s")
+    (babashka . "bb socket-repl %s")))
+
+;;;###autoload
+(defun inf-clojure-socket (&optional &rest args)
```

Comment:
I'd probably name this `inf-clojure-socket-repl` as it's a more descriptive name. 

I'm wondering why you didn't list out the supported params explicitly and made it possible for the user to be prompted about things like `host` and `port` to use.

Reply:
I've updated the name, thanks for this.

I've also changed the function signature to be `cmd` and operate essentially the same as `inf-clojure`. Choosing a port can be done with the custom variable `inf-clojure-socket-repl-port` but otherwise it's an implementation detail and gets automatically chosen. The user will be prompted for a command to use (from the defined defaults in `inf-clojure-socket-repl-startup-forms`) just like the main function too!

I did originally use the &rest/plist approach because there were a lot of variables that I thought users may want to override but now that the function is simpler, hopefully this is a good change.